### PR TITLE
Update compat matrix in kafka output plugin docs to mention SASL

### PR DESCRIPTION
**Pull request by dedemorton**
@acchen97 @suyograo Changes for elastic/logstash#6489.

### CODE REVIEW

**Comment by dedemorton** [comment]
@suyograo @acchen97 Check out these changes. Not sure if we are 100% there, but I think it's getting clearer...

File: lib/logstash/outputs/kafka.rb [hunk 1]
```diff
@@ -11,18 +11,26 @@
...
-# |0.9       |2.4.x - 5.x.x   | 4.x.x |SSL |Works with the new getter/setter APIs (`event.set('[product][price]', 10)`)
-# |0.10.0.x  |2.4.x - 5.x.x   | 5.x.x |SSL |Not compatible with the <= 0.9 broker
-# |0.10.1.x  |2.4.x - 5.x.x   | 6.x.x |SSL |Not compatible with <= 0.10.0.x broker
+# |Kafka Client Version |Logstash Version |Plugin Version |Why?
+# |0.8       |2.0.0 - 2.x.x   |<3.0.0 |Legacy, 0.8 is still popular 
+# |0.9       |2.0.0 - 2.3.x   | 3.x.x |Works with the old Ruby Event API (`event['product']['price'] = 10`)
+# |0.9       |2.4.x - 5.x.x   | 4.x.x |Works with the new getter/setter APIs (`event.set('[product][price]', 10)`)
+# |0.10.0.x  |2.4.x - 5.x.x   | 5.x.x |Not compatible with the <= 0.9 broker
+# |0.10.1.0  |2.4.x - 5.x.x   | 6.0.0 - 6.1.1 |Not compatible with <= 0.10.0.x broker
+# |0.10.1.1  |2.4.x - 5.x.x   | 6.1.2 - 6.x.x |Not compatible with <= 0.10.1.0 broker
```
Comment (dedemorton) [inline lib/logstash/outputs/kafka.rb]: I'm not sure about this compatibility statement. Added the row and followed the convention in the balte.

Comment (suyograo) [inline lib/logstash/outputs/kafka.rb]: Same as input. Remove this line

**Comment by suyograo** [comment]
Remove the last line in compat matrix and good to go.

**Pull request closed without merging**